specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/workspaceRP.json

```diff
@@ -3356,7 +3435,7 @@
           }
         },
         "managedNetwork": {
-          "$ref": "#/definitions/ManagedNetworkSettings"
+          "$ref": "#/definitions/ManagedNetworkSettingsEx"
```

Comment:
>ManagedNetworkSettingsEx [](http://example.com/codeflow?start=33&length=24)

I'm not clear why you are doing it this way, as opposed to just adding properties on ManagedNetworkSettings. Want to explain?

Reply:
reverted

Reply:
this is generated based on the code, I guess the engineer just consider this an easy way to deserialize to a derived type for optional fields